doc/source/user_guide/codelist.rst

```diff
@@ -0,0 +1,209 @@
+.. _codelist:
+
+Codelist
+========
+
+A "codelist" (named in reference to code lists in `SDMX <https://sdmx.org/>`_) is a yaml
+file that contains a list of allowed values for any **index** dimension of the IAMC
+format (model, scenario, region, variable). Which index dimension a code list will
+be applied to is determined by the **name of the folder** in which it is located. For
+example, if a codelist is located in a folder called ``model/`` it will be applied to
+the "model" index dimension. 
+
+A codelist must only contain a single list of allowed values. In the most simple case
+the list items are strings, e.g.:
+
+.. code:: yaml
+
+  - allowed_value_a
+  - allowed_value_b
+  - allowed_value_c
+  - ...
+
+In case of the "model" or "scenario" index dimension, such a simple list sufficient.
+More details on 
```

Comment:
This sentence is not complete. However, I would omit this entire paragraph. Users will scroll through the page for the heading that they are interested in (variables, regions, ...).

Reply:
Sounds good, removed it.